File: archive.py
from os import walk
import shutil
from pathlib import Path


def is_txt(s):
    return '.txt' in s


base_addr = '.'
_, _, file_names = next(walk(base_addr))
poems = list(filter(is_txt, file_names))

# print(poems)

for p in poems:
    typ = p.split('-')[0]
    school = p.split('-')[1]
    author = p.split('-')[2]
    name = p.split('-')[3][:-4]

    # print(typ, school, author, name)

    dst_addr = base_addr+'/'+typ+'/'+school+'/'+author
    Path(dst_addr).mkdir(parents=True, exist_ok=True)
    shutil.move(base_addr+'/'+p, dst_addr+'/'+p)

File: util/archive.go
package util

import (
	"fmt"
	"os"
	"strings"
)

func Archive(baseAddr string, dstFolder string, poems []string) {
	for _, v := range poems {
		attributes := strings.Split(v, "-")
		typ := attributes[0]
		school := attributes[1]
		author := attributes[2]
		name := attributes[3][0 : len(attributes[3])-4]
		fmt.Println(typ, school, author, name)
		dstAddr := dstFolder + "/" + typ + "/" + school + "/" + author
		os.MkdirAll(dstAddr, 0766)
		// os.Rename(baseAddr+"/"+v, dstAddr+"/"+v)
		CopyFile(baseAddr+"/"+v, dstAddr+"/"+v)
	}
}
